lib/Model/Pass/DeduplicateEquivalentTypes.cpp

```diff
@@ -221,9 +221,9 @@ class TypeSystemDeduplicator {
   }
 
 private:
-  void addEdge(const model::TypeDefinition *T, const model::QualifiedType &QT) {
-    auto *DependantType = QT.UnqualifiedType().get();
-    TypeToNode.at(T)->addSuccessor(TypeToNode.at(DependantType));
+  void addEdge(const model::TypeDefinition &T, const model::Type &Type) {
+    if (const model::TypeDefinition *Definition = Type.skipToDefinition())
+      TypeToNode.at(&T)->addSuccessor(TypeToNode.at(Definition));
   }
```

Comment:
Eventually we should try to push building the graph of dependencies between `TypeDefinition`s into `model::Binary` and stop exposing `skipToDefinition` altogether.
But not today.